### DISCUSSION TITLE
Newbie needs support to setup his enocean devices

### DISCUSSION THREAD

**Discussion by mectiv** [mak-gitdev/HA_enoceanmqtt#48] [Q&A]
Hi!
I recently switched from openhub to HomeAssistant. Due to some threads on the HA forum, I became aware of HA enoceanmqtt. I successfully installed the addon and tried to add my first rocker switch. Unfortunately, I don't get it running. This is probably because I am lacking know-how about HA. But maybe someone can still help me. 
In my house, I got three enocean devices: lights, shutters, and temp sensors. The lights and shutters are obviously just physical switches with different control mechanisms. I added a two switches (one for light, one for shutter) to my configuration:
 
`[FlurEG]
address         = 0xFFD7BE85
rorg            = 0xF6
func            = 0x02
type            = 0x02
sender          = 0xFFD5C105
virtual         = 1

[WZNord]
address         = 0xFFD7BE9E
rorg            = 0xD2
func            = 0x05
type            = 0x00
sender          = 0xFFD5C11E
virtual         = 1`

Unfortunately I just see this on my dashboard:
<img width="382" alt="image" src="https://user-images.githubusercontent.com/7467091/221029843-5dd0a7e7-72b5-49fa-adc8-cc26eab877c0.png">

What am I doing wrong?
Thank you!

Kind regards,
Christian

**Comment by mak-gitdev** [comment]
Hi @mectiv,

> I added a two switches (one for light, one for shutter) to my configuration

Do you want to add physical or virtual switches ?

> 
> ```
>[FlurEG]
>address = 0xFFD7BE85
>rorg = 0xF6
>func = 0x02
>type = 0x02
>sender = 0xFFD5C105
>virtual = 1
>
>[WZNord]
>address = 0xFFD7BE9E
>rorg = 0xD2
>func = 0x05
>type = 0x00
>sender = 0xFFD5C11E
>virtual = 1

Where do these addresses come from ?
Also D2-05-00 is not a switch but a roller shutter.

**Comment by mectiv** [comment]
Hi mak-gitdev,

thanks for your reply. I appriciate your help.

> Do you want to add physical or virtual switches ?

I would like to control the light through HA and also see the current status (on/off).So I think I need both. 

> Where do these addresses come from ?

The value of "address" is the ID of the physical switch. The "sender" is in the range of my USB300 Stick and is programmed in the enocean enviroment to control the lights/shutters.

Thanks
Christian

**Comment by mak-gitdev** [comment]
@mectiv,

I think you may have misunderstood how this works.
Let's start from the beginning so I can better help you.
Could you describe your EnOcean setup ? What are the EnOcean devices you are using to control your lights and shutters ?

**Comment by mectiv** [comment]
No problem, I will try to describe it :)

We have physical switches in the rooms that control lights as well as shutters. These are from type enocean PTM215.
Also, we have temperature sensors from eltako (FTR55H).

In the electrical control cabinet there is everything made by eltako:
- FAM14
- FSR14-4x
- FSB14
- FSNT14-12V/12W (was added because the power unit of the FAM14 is broken)
- FHK14

**Comment by commentsonly** [comment]
@mectiv
replace: address = 0xFFD7BE85 with : 0xFFFFFFFF
Since you got a FAM14/FSB14 I bet you have the PCT14 Eltako Software? If yes, use it's mode where it detects button presses.
If you configured the addon correctly the PCT14 software should see the telegrams sent from the virtual button. (it does send them if you select somthing differnt, AO or AI from the selector box you posted in the your primary post.
Teach in that virtual button besides the physical one on the PCT14 software and you can run the shutter from physical and the virtual button from within HA.

Ask in the home asssistant community forum, I'd then pm you using german lang ... likely easier. Or simply wait for the solution mak-gitdev comes up with ...

**Comment by mak-gitdev** [comment]
@commentsonly,

Thank you very much for proposing your help to @mectiv.
I think it is a good idea as you have similar setups.

Just @mectiv, the principle in your case is that you can add virtual buttons in HA to control your devices, just like you would have done with physical rocker switches.
There is a possibility to directly control the Eltako devices without using virtual buttons but it is not yet available.
See this discussion for more details.

So I let @commentsonly guiding you to use virtual buttons to control your Eltako devices.

**Comment by commentsonly** [comment]
@Marc ... unsure if it works but I do think I know where **mectiv** did take the wrong exit. He likely assumed that a "virtual" button acts identical to an existing physical button.
And well, if thinking about it this would be a real gaol.
Assume you have 10 shutters, 10 lights and all teached in with 20 physical buttons.
If defining virtual buttoins in their current form means you have to teach in all these virtual ones to the wanted receiving devices. Sure that's the propper way and follows what EnOcean implemented.
But I remember I once had a networkcard where I was entitled to change the MAC to the id I wanted it to be. Yes if doing so and not being accurate one ends up with 2 network cards having the same ID which then results in a bigger problem.

What If I would define a virtual EnOcean button strictly identical to what a physical button is at my home?
Sort of my gira button is 0xFEE50F9C. And it's teached into my FAM14/FSB14 to drive a shutter.
If I would now be able to define a virtual one using that same ID ... there's no need to teach in anothrer button to the FAM/FSB combination since that one won't bother if a telegram comes from the physical or the virtual button. Possible?
I mean is that "stickid + range" something you hardcoded, or is that something the stick it fixed to?

And yes I try to help mectvc if I'm able. I works like a charm by now. With only 1 script for 12 shutters. 😄

**Comment by mak-gitdev** [comment]
This would be nice but unfortunately this is not possible.
Not because of some tricks by myself but because of EnOcean itself.
Each EnOcean device has a unique id hardcoded to the device. This is the EnOcean Unique Radio ID ([EURID](https://www.enocean-alliance.org/EURID/)) of the device and it cannot be replicated by your EnOcean transceiver.
In addition to the EURID, your EnOcean transceiver has 128 alternative addresses (Base ID) which are in the range 0xFF800000-0xFFFFFFFE in steps of 128.
This Base ID can instead be modified.

**Reply by commentsonly** [reply]
@mectiv
quick solution
try this one, assuming the sender is as if described in the wiki
```
[FlurEG]
address = 0xFFFFFFFF
rorg = 0xF6
func = 0x02
type = 0x02
sender = 0xFFD5C105
virtual = 1
```
create a button card in the lovelac frontend of home assistant
in Homeassistant>Overview>... in the top right corner>Edit Dashboard>+Add Card>Button
Entity could remain blank for the moment.
Tap action: Call Service
Serivice: select.select_option (likely turns into 'Select: Select' after you entered it.
Target: select one of the 'e2m_......' devices you find. There should be one in the form 'e2m_flureg_action'
In the option field enter 'AO' or 'AI'.

You could have a look at the YAML code pressing SHOW CODE EDITOR.
Short explanation:
You call a service, which is the selector (the one you shared the screenshot from your first post)
the entity_id is that selector, eg. the virtual button you defined.
And the option is the individual button. AO = left upper, AI, left lower, BO right upper, BI right lower button if a F6-02-02.

As said earlier, this is a simple solution, reality is more complexed, since the physical buttons send a telegram when being pressed and another one if being released.

To do that you need some scripting or an automation or the custom:button-card since you need to fire multiple services.
Eg the above service with for example AO followed by the same service with option None.
But AFAIK it won't matter if you use AO to switch something on and AI to switch it off. It only becomes important if you want an action taking place as long as you keep a button pressed until it gets released. It all depends upon what you want to do.
